warehouse/oidc/tasks.py

```diff
@@ -0,0 +1,59 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from warehouse import tasks
+from warehouse.metrics import IMetricsService
+from warehouse.packaging.models import File, Project, Release
+
+
+@tasks.task(ignore_result=True, acks_late=True)
+def compute_oidc_metrics(request):
+    metrics = request.find_service(IMetricsService, context=None)
+
+    projects_configured_oidc = (
+        request.db.query(Project.id).distinct().join(Project.oidc_publishers)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** ‏I'm having some difficulty in getting this query to run in a shell:

```shellsession
$ make shell
>>> from warehouse.packaging.models import File, Project, Release
>>> db.query(Project.id).distinct().join(Project.oidc_publishers)
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
AttributeError: type object 'Project' has no attribute 'oidc_publishers'
```

I'm not entirely certain why this relationship isn't being expressed. 🤔 

Were you able to run this query?

Reply:
Never mind, had to call `sqlalchemy.orm.configure_mappers()` after import to create the mappings, something that happens automatically in production.

Reply:
Glad you figured that out, I wouldn't have been much help there :P All I had was "unit tests seem to work!"